code/modules/reagents/chemistry/reagents/drugs.dm

```diff
@@ -479,31 +546,51 @@
 		update_flags |= M.adjustToxLoss(1, FALSE)
 	return ..() | update_flags
 
+//makes you resistant to stuns and stamina damage, heals a small amount of stamina damage, causes a large amount of toxin damage, on removal forces you to throw up blood.
 /datum/reagent/aranesp
 	name = "Aranesp"
 	id = "aranesp"
 	description = "An illegal performance enhancing drug. Side effects might include chest pain, seizures, swelling, headache, fever... ... ..."
 	reagent_state = LIQUID
 	color = "#60A584" // rgb: 96, 165, 132
 	taste_description = "bitterness"
+	addiction_chance = 5
+	addiction_threshold = 5
+	addiction_decay_rate = 0.2
+	/// how much do we edit the stun and stamina mods? lower is more resistance
+	var/tenacity = 0.7
+
+/datum/reagent/aranesp/on_mob_add(mob/living/L)
+	if(ishuman(L))
+		var/mob/living/carbon/human/H = L
+		H.physiology.stun_mod *= tenacity
+		H.physiology.stamina_mod *= tenacity
 
 /datum/reagent/aranesp/on_mob_life(mob/living/M)
 	var/update_flags = STATUS_UPDATE_NONE
-	update_flags |= M.adjustStaminaLoss(-40, FALSE)
-	if(prob(90))
-		update_flags |= M.adjustToxLoss(1, FALSE)
+	var/recent_consumption = holder.addiction_threshold_accumulated[type]
+
+	// small amount of stam regen, meaning that if you can buy time between hits
+	// you can take 6 disabler shots (30 base * 0.7 stam mod = 21 stam damage each, 5 shots to 105 which is stamina crit. but you can heal 6 in 3 cycles) before going down
+	update_flags |= M.adjustStaminaLoss(-2, FALSE)
+	if(prob(10 * DRAWBACK_CHANCE_MODIFIER(recent_consumption)))
+		update_flags |= M.adjustToxLoss(4, FALSE)
 	if(prob(5))
 		M.emote(pick("twitch", "shake", "tremble","quiver", "twitch_s"))
-	var/high_message = pick("really buff", "on top of the world","like you're made of steel", "energized", "invigorated", "full of energy")
 	if(prob(8))
-		to_chat(M, "<span class='notice'>[high_message]!</span>")
-	if(prob(5))
-		to_chat(M, "<span class='danger'>You cannot breathe!</span>")
-		update_flags |= M.adjustOxyLoss(15, FALSE)
-		M.Stun(2 SECONDS)
-		M.AdjustLoseBreath(2 SECONDS)
+		var/high_message = pick("really buff", "on top of the world","like you're made of steel", "energized", "invigorated", "full of energy")
+		to_chat(M, "<span class='notice'>You feel [high_message]!</span>")
 	return ..() | update_flags
 
+/datum/reagent/aranesp/on_mob_delete(mob/living/M)
+	if(ishuman(M))
+		var/mob/living/carbon/human/H = M
+		H.physiology.stun_mod /= tenacity
+		H.physiology.stamina_mod /= tenacity
+		H.vomit(blood = TRUE, stun = FALSE)
+		H.adjustToxLoss(10)
```

Comment:
now THIS one is not visual at all.
Can you check that anaresp patches don't straight murder you?

Reply:
![image](https://user-images.githubusercontent.com/69320440/176259632-b4f40373-26d9-4d12-bfa8-cdc54b67c5b9.png)

I hate patches.

Reply:
editted to be 10 seconds of losebreath, which won't stack up. will cause 25 damage, but then will heal.